ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -112,7 +112,9 @@ public static SourceBuilder instance() {
       // we do not need to force a materialization at this source step since the
       // re-partitioned topic will be used for any subsequent state stores, in lieu
       // of the original source topic, thus avoiding the issues above.
-      // See https://github.com/confluentinc/ksql/issues/6650
+      // See https://github.com/confluentinc/ksql/issues/6650.
+      // This intermittent transformValues() call is unneccessary in this case,
```

Comment:
```suggestion
      // This intermittent transformValues() call is unneccessary in this case,
      // i.e., it could be combined with the subsequent transformValues() call into a single transformValues() call,
```